### PR TITLE
[14.0][IMP] shopfloor: imp cancel line/package in location content transfer

### PR DESCRIPTION
**Pull request by TDu**
ref.: rau-229

### CODE REVIEW

File: shopfloor/services/location_content_transfer.py [hunk 1]
```diff
@@ -913,15 +921,21 @@ def stock_out_line(self, location_id, move_line_id):
         # We need to set qty_done at 0 because otherwise
         # the move_line will not be deleted
         move_line.qty_done = 0
-        move._do_unreserve()
-        move._recompute_state()
+        if self.is_allow_move_create() and self.env.user == move.picking_id.create_uid:
+            # Owned by the user deleting the move
+            move._do_unreserve()
+            move._recompute_state()
+            move._action_cancel()
```
Comment (jbaudoux) [inline shopfloor/services/location_content_transfer.py]: Looks good to call `_action_cancel` before before creating the stock issue.
`_action_cancel` is already doing the `_do_unreserve`
```suggestion
            move._action_cancel()
```

Also update docstring

File: shopfloor/services/location_content_transfer.py [hunk 2]
```diff
@@ -855,17 +855,25 @@ def stock_out_package(self, location_id, package_level_id):
...
-            package_move._recompute_state()
+            package = package_level.package_id
+            if (
+                self.is_allow_move_create()
+                and self.env.user == package_move.picking_id.create_uid
+            ):
+                # Owned by the user deleting the move
+                package_move._do_unreserve()
+                package_move._recompute_state()
+                package_move._action_cancel()
```
Comment (jbaudoux) [inline shopfloor/services/location_content_transfer.py]: Looks good to call `_action_cancel` before before creating the stock issue.
`_action_cancel` is already doing the `_do_unreserve`
```suggestion
            move._action_cancel()
```

Also update docstring

File: shopfloor/services/location_content_transfer.py [hunk 3]
```diff
@@ -855,17 +855,25 @@ def stock_out_package(self, location_id, package_level_id):
...
+            else:
+                # Not owned only unassign the user
+                stock = self._actions_for("stock")
+                stock.unmark_move_line_as_picked(package_move.move_line_ids)
```
Comment (jbaudoux) [inline shopfloor/services/location_content_transfer.py]: We also need to unreserve before creating the stock issue.

File: shopfloor/services/location_content_transfer.py [hunk 4]
```diff
@@ -913,15 +921,21 @@ def stock_out_line(self, location_id, move_line_id):
...
-        move._recompute_state()
+        if self.is_allow_move_create() and self.env.user == move.picking_id.create_uid:
+            # Owned by the user deleting the move
+            move._do_unreserve()
+            move._recompute_state()
+            move._action_cancel()
+        else:
+            # Not owned only unassign the user
+            stock = self._actions_for("stock")
+            stock.unmark_move_line_as_picked(move.move_line_ids)
```
Comment (jbaudoux) [inline shopfloor/services/location_content_transfer.py]: We also need to unreserve before creating the stock issue.

**Review by jbaudoux — CHANGES_REQUESTED** [review]

**Review by jbaudoux — APPROVED** [review]
LGTM

**Review by mt-software-de — APPROVED** [review]
LG

**Comment by jbaudoux** [comment]
@Tdu Rebase & squash fixup

**Comment by TDu** [comment]
@jbaudoux Squashed and rebased

**Pull request merged**